gtests/net/mptcp/sockopts/sockopt_maxseg_mptcp.pkt

```diff
@@ -0,0 +1,29 @@
+--tolerance_usecs=100000
+`../common/defaults.sh`
+
+0.0  socket(..., SOCK_STREAM, IPPROTO_MPTCP) = 3
++0   getsockopt(3, SOL_TCP, TCP_MAXSEG, [536], [4]) = 0
+
++0   setsockopt(3, SOL_TCP, TCP_MAXSEG, [1460], 4) = 0
++0   getsockopt(3, SOL_TCP, TCP_MAXSEG, [1460], [4]) = 0
+
++0   bind(3, ..., ...) = 0
++0   listen(3, 1) = 0
+
++0     <  S   0:0(0)          win 8000  <mss 1460, sackOK, nop, nop, nop, wscale 0, mpcapable v1 flags[flag_h] nokey>
++0     >  S.  0:0(0)  ack 1             <mss 1460, nop, nop, sackOK, nop, wscale 8, mpcapable v1 flags[flag_h] key[skey]>
++0.01  <   .  1:1(0)  ack 1   win 8000                                             <mpcapable v1 flags[flag_h] key[ckey=2, skey]>
++0   accept(3, ..., ...) = 4
+
++0   getsockopt(4, SOL_TCP, TCP_MAXSEG, [1460], [4]) = 0
++0   write(4, ..., 1250) = 1250
+
++0     >  P.  1:1251(1250)    ack 1     <dss dack4=1   dsn8=1 ssn=1 dll=1250 nocs, nop, nop>
++0.01  <   .  1:1(0)          ack 1251  win 256  <dss dack4=1251 nocs>
+
++0   setsockopt(4, SOL_TCP, TCP_MAXSEG, [800], 4) = 0
++0   getsockopt(4, SOL_TCP, TCP_MAXSEG, [1460], [4]) = 0
++0   write(4, ..., 1250) = 1250
+
++0     >  P.  1251:2501(1250) ack 1     win 1050 <dss dack4=1   dsn8=1251 ssn=1251 dll=1250 nocs, nop, nop>
++0.01  <   .  1:1(0)          ack 2501  win 256  <dss dack4=1251 nocs>
```

Comment:
just not to forget about that: please create a new subflow here by injecting `<` a SYN+MPJ, and then making sure the kernel generates (`>`) a SYN+ACK+MPJ with the expected MSS, something similar to what is done in `mp_join_server.pkt`:

https://github.com/multipath-tcp/packetdrill/blob/94a68e1b94ca568e528524598e9136739f417fc4/gtests/net/mptcp/mp_join/mp_join_server.pkt#L25-L26

You will also need to add this in the SYN+MPC above `addr[caddr0] > addr[saddr0]`, see:

https://github.com/multipath-tcp/packetdrill/blob/94a68e1b94ca568e528524598e9136739f417fc4/gtests/net/mptcp/mp_join/mp_join_server.pkt#L14